src/main/java/de/tum/in/www1/artemis/domain/tutorialgroups/TutorialGroup.java

```diff
@@ -62,6 +62,24 @@ public class TutorialGroup extends DomainObject {
     @JsonIgnoreProperties(value = "tutorialGroup", allowSetters = true)
     private Set<TutorialGroupRegistration> registrations = new HashSet<>();
 
+    // ==== Transient fields ====
+
+    @Transient
+    @JsonSerialize
+    private Boolean isUserRegistered;
+
+    @Transient
+    @JsonSerialize
+    private Integer numberOfRegisteredUsers;
+
+    @Transient
+    @JsonSerialize
+    private String teachingAssistantName;
+
+    @Transient
+    @JsonSerialize
+    private String courseTitle;
```

Comment:
When is this needed? Can you not re-use the course object in the client?

Reply:
I like it because it makes certain parts much cleaner. For example then I do not have to pass the course separately as an extra `@Input` to sub components just for displaying the course title. Is keeping it in a hard blocker for you?

Reply:
For example on the detail page

Reply:
I understand your reasoning and the positive sides, but I don't think we have done this so far. There are also some downsides, like cluttering the models with these attributes. Thus, I'd be a design change and more reviewers should add their opinion on this. 🙂 
@JohannesStoehr @manuelmanso What do you say?

Reply:
The other clean option would be a DTO but we were in the past hesitant to introduce too many DTOS.